changelog.d/11827.bugfix

```diff
@@ -0,0 +1 @@
+Ensure we always log HTTP status codes as integers instead of e.g. `HTTPStatus.OK`.
```

Comment:
For consistency with other bugfix changelogs

```suggestion
Fix a bug introduced in Synapse 0.33.3 causing requests to sometimes log strings such as `HTTPStatus.OK` instead of integer status codes.
```